### PR TITLE
Fixes to states/virt.py

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Adding the ability to configure some options when generating libvirt keys, particularly the number of days that the certificates will be valid for.

### What issues does this PR fix or reference?
#41667 

### Previous Behavior
Previously options were unavailable when generating certificates for use with libvirt.

### New Behavior
This change allows options including the number of days that the certificate will be valid for to be specified.

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/pillar/libvirt.py [hunk 1]
```diff
@@ -91,8 +99,10 @@ def gen_hyper_keys(minion_id,
         with salt.utils.fopen(srvinfo, 'w+') as fp_:
             infodat = ('organization = salted\ncn = {0}\ntls_www_server'
                        '\nencryption_key\nsigning_key'
-                       '\ndigitalSignature').format(
-                               __grains__['fqdn'])
+                       '\ndigitalSignature\n'
+                       'expiration_days = {1}'
+                       ).format(
+                               __grains__['fqdn'], expiration_days)
```
Comment (isbm) [inline salt/pillar/libvirt.py]: This line can be on the previous though. But maybe multi-line template would work better here?

File: salt/states/virt.py [hunk 2]
```diff
@@ -52,20 +53,53 @@ def keys(name, basepath='/etc/pki'):
...
-    #libvirt.serverkey.pem
-    #libvirt.servercert.pem
-    #libvirt.clientkey.pem
-    #libvirt.clientcert.pem
-    #libvirt.cacert.pem
+    # libvirt.serverkey.pem
+    # libvirt.servercert.pem
+    # libvirt.clientkey.pem
+    # libvirt.clientcert.pem
+    # libvirt.cacert.pem
```
Comment (isbm) [inline salt/states/virt.py]: Needed?

File: salt/states/virt.py [hunk 3]
```diff
@@ -52,20 +53,53 @@ def keys(name, basepath='/etc/pki'):
...
 
     ret = {'name': name, 'changes': {}, 'result': True, 'comment': ''}
-    pillar = __salt__['pillar.ext']({'libvirt': '_'})
+
+    # Grab all kwargs to make them available as pillar values
+    # rename them to something hopefully unique to avoid
+    # overriding anything existing
+    pillar_kwargs = {}
+    for (key, value) in kwargs.iteritems():
```
Comment (isbm) [inline salt/states/virt.py]: Parenthesis is unnecessary here.

**Review by isbm — CHANGES_REQUESTED** [review]
Could you please look at these changes?

**Comment by isbm** [comment]
> ## Tests written?
> No

Why?

File: salt/pillar/libvirt.py (hunk 1 above)
Comment (garethgreenaway) [inline salt/pillar/libvirt.py]: Fair.  Seemed more readable at the time to have it on it's own line but moving it up one is fine as well.

File: salt/states/virt.py (hunk 2 above)
Comment (garethgreenaway) [inline salt/states/virt.py]: Not needed, but much like the line too long warnings are caught via the lint plugin I'm using so they show up as errors/warnings.  We're not failing on them but it makes for more readable code so I tend to fix them when I find them.

File: salt/states/virt.py (hunk 3 above)
Comment (garethgreenaway) [inline salt/states/virt.py]: Good catch, thanks.  This was left over from something else.

File: salt/states/virt.py (hunk 2 above)
Comment (isbm) [inline salt/states/virt.py]: I mean, maybe just yank this comment at all? Why garbage in the code...

**Review by isbm — CHANGES_REQUESTED** [review]
Maybe still consider to remove these comments about keys. They are anyways in the code.

Very much appreciated the unit test! Your karma is up. :wink:

File: salt/states/virt.py (hunk 2 above)
Comment (cachedout) [inline salt/states/virt.py]: I agree with @isbm here. Seems like these should just be removed.

**Comment by cachedout** [comment]
@isbm How is this looking to you now?

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

**Comment by isbm** [comment]
@cachedout Yep, now I can stop bothering your new employee. :laughing:
@garethgreenaway thanks!

**Review by isbm — APPROVED** [review]

**Comment by garethgreenaway** [comment]
@isbm No worries!  Thanks for your feedback.

**Comment by cachedout** [comment]
@isbm We're always glad for your help and feedback. Thanks. :]

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

**Comment by garethgreenaway** [comment]
Go Go Jenkins!

**Review by rallytime — APPROVED** [review]
Very nice.

**Pull request merged**